linkml/transformers/rollup_transformer.py

```diff
@@ -0,0 +1,133 @@
+        self,
+        target_class: str,
+        config: Optional[FlattenTransformerConfiguration] = None,
+    ):
+        super().__init__()
+        self.target_class = target_class
+        self.config = config or FlattenTransformerConfiguration()
+        self._descendant_classes: set[str] = set()
+        self._class_to_slot_map: dict[str, set[str]] = {}
+        self._collect_slots_for_class: dict[str, list[SlotDefinition]] = {}
+
+    def transform(self, tgt_schema_name: str = None, top_class: ClassDefinitionName = None) -> SchemaDefinition:
+        """
+        Transform the schema by flattening the inheritance hierarchy.
+
+        Args:
+            tgt_schema_name: Optional target schema name (not used in this transformer)
+            top_class: Optional top class name (not used in this transformer)
+
+        Returns:
+            A transformed SchemaDefinition with the target class having all slots from its descendants.
+        """
+        if not self.schemaview:
+            raise ValueError("SchemaView not set. Ensure schema is set by calling set_schema().")
+
+        schema_copy = copy.deepcopy(self.source_schema)
+        view = SchemaView(schema_copy)
+        descendants = view.class_descendants(self.target_class, imports=True, mixins=True, reflexive=False, is_a=True)
+        self._descendant_classes = set(descendants)
+        self._collect_slots(view)
+        return self._create_flattened_schema(schema_copy, view)
+
+    def _collect_slots(self, view: SchemaView) -> None:
+        """
+        Collect all slots from the target class and its descendant classes.
+        Populates a mapping of class name to the set of its slot names.
+        """
+        # Get induced slots for the target class.
+        target_slots = view.class_induced_slots(self.target_class)
+        self._collect_slots_for_class[self.target_class] = target_slots
+
+        for descendant in self._descendant_classes:
+            slots = view.class_induced_slots(descendant)
+            self._collect_slots_for_class[descendant] = slots
+            self._class_to_slot_map[descendant] = {s.name for s in slots}
+
+    def _create_flattened_schema(self, schema: SchemaDefinition, view: SchemaView) -> SchemaDefinition:
+        """
+        Create a new flattened version of the schema based on collected slots.
+        """
+        target_class_def = schema.classes[self.target_class]
+        existing_slot_names = set(target_class_def.slots or [])
+        all_slots = set()
+
+        for descendant in self._descendant_classes:
+            if descendant in schema.classes:
+                for slot_name in self._class_to_slot_map.get(descendant, set()):
+                    if slot_name not in existing_slot_names:
+                        all_slots.add(slot_name)
+
+        #!
+        if target_class_def.slots:
+            target_class_def.slots.extend(list(all_slots))
+        else:
+            target_class_def.slots = list(all_slots)
```

Comment:
This is personal taste, but I prefer doing

```py
if not target_class_def.slots:
    target_class_def.slots = []

target_class_def.slots.extend(list(all_slots))
```

I find it a little simpler/clearer but YMMV.